api/client/client.go

```diff
@@ -171,6 +169,37 @@ func (c *Client) RemoveAppToken(ctx context.Context, tokenID string, auth Auth)
 	return err
 }
 
+// AddFolder adds a folder by name.
+func (c *Client) AddFolder(ctx context.Context, projID, name string, public bool, auth Auth) (*pb.AddFolderReply, error) {
```

Comment:
Here, `public` indicates that the folder should be viewable over the public gateway. Currently, not handled.